forte/data/readers/conll03_reader_new.py

```diff
@@ -0,0 +1,145 @@
+                conll_components = line.split()
+
+                word = conll_components[0]
+                pos = conll_components[1]
+                chunk_id = conll_components[2]
+
+                ner_tag = conll_components[3]
+
+                # A new ner tag occurs.
+                if ner_tag == "O" or ner_tag.split("-")[0] == "B":
+                    # Add previous ner tag to sentence if it exists.
+                    if prev_y is not None:
+                        entity_mention = EntityMention(pack,
+                                                start_index, offset - 1)
+                        entity_mention.ner_type = prev_y
+
+                    # Start process current ner tag.
+                    if ner_tag == "O":
+                        # Current ner tag is O, reset information.
+                        prev_x = None
+                        prev_y = None
+                        start_index = -1
+                    else:
+                        # Current ner tag is B.
+                        prev_x = "B"
+                        prev_y = ner_tag.split("-")[1]
+                        start_index = offset
+                # This ner tag is connected to previous one.
+                else:
+                    x, y = ner_tag.split("-")
+                    assert x == "I", "Unseen tag %s in the file." % x
+                    assert y == prev_y, "Error in %s." % ner_tag
+                    assert prev_x in ("B", "I"), "Error in %s." % ner_tag
+                    prev_x = "I"
+
+                word_begin = offset
+                word_end = offset + len(word)
+
+                # Add tokens.
+                token = Token(pack, word_begin, word_end)
+                token.pos = pos
+                token.chunk = chunk_id
+
+                text += word + " "
+                offset = word_end + 1
+                has_rows = True
+            else:
+                if not has_rows:
+                    # Skip consecutive empty lines.
+                    continue
+                # Add sentence
+                Sentence(pack, sentence_begin, offset - 1)
+
+                # Handle the last ner tag if exists.
+                if prev_x is not None:
+                    entity_mention = EntityMention(pack, start_index,
+                                                    offset - 1)
+                    entity_mention.ner_type = prev_y
+
+                # Reset information.
+                sentence_cnt += 1
+                has_rows = False
+                prev_y = None
+                prev_x = None
+                sentence_begin = offset
+
+        if has_rows:
+            # Add the last sentence if exists.
+            Sentence(pack, sentence_begin, offset - 1)
+            sentence_cnt += 1
+
+        pack.set_text(text, replace_func=self.text_replace_operation)
+
+        Document(pack, 0, len(text))
+
+        pack.pack_name = collection
+        doc.close()
+
+        yield pack
```

Comment:
Currently, an entire dataset file will be parsed into a single datapack and there is only one big `Document` entry. However, there are lines in the dataset that are denoting document start (`-DOCSTART- -X- -X- O`) which means it is possible that multiple documents exist in a single dataset file. Here are two ways to go:

* Keep a single datapack but need to add multiple documents based on `-DOCSTART- -X- -X- O`. `yield` this datapack when the entire dataset file is parsed.

* Keep multiple datapacks and each datapack corresponding to a single document in a dataset file. `yield` the datapack whenever a document is parsed in a dataset file.

Reply:
Both ways are fine in this example. Having multiple datapacks will help in the long run.

Reply:
Ok. I will use one pack for one DOC, since it will be more clear.

Reply:
I am considering some possible problem if I make one pack for one document.
1. The previous existing Coll03 dataset does not contain `-DOCSTART-`. Missing this line might cause some corner case for the implementation.
2. Each pack has  a name, and current the name is the file name. If I create several pack for one file, I might need to make the pack name as `filename-pack%d`. This could break some assumption?

Reply:
For 1, you can do some check and make it work both cases.
For 2, there is not a lot assumptions on the file name, afaik

Reply:
I update the code, so that each Document should be mapped to one pack.